app/assets/javascripts/discourse/app/controllers/discovery-sortable.js

```diff
@@ -27,17 +29,38 @@ export const queryParams = {
 const controllerOpts = {
   discoveryTopics: controller("discovery/topics"),
   queryParams: Object.keys(queryParams),
+
+  @discourseComputed(...Object.keys(queryParams))
+  queryString() {
+    let paramStrings = [];
+
+    this.queryParams.forEach((key) => {
+      if (this[key]) {
+        paramStrings.push(`${key}:${this[key]}`);
+      }
+    });
+
+    return paramStrings.join(" ");
+  },
+
+  @action
+  updateTopicsListQueryParams(queryString) {
+    for (const match of queryString.matchAll(/(\w+):([^:\s]+)/g)) {
```

Comment:
My comment is a bit premature, but I feel that we need better separator for options than whitespace, maybe something like: "|". I am thinking about that because future options may consist spaces like:
`category: fun facts|status: open`

Reply:
I think it is unlikely at this point given the specification we have right now for the query language but we can easily make changes to the regexp in the future.